In [5]:
%pip install lxml


[notice] A new release of pip is available: 24.0 -> 24.2
[notice] To update, run: pip install --upgrade pip
Note: you may need to restart the kernel to use updated packages.


In [6]:
%pip install MyCapytain


[notice] A new release of pip is available: 24.0 -> 24.2
[notice] To update, run: pip install --upgrade pip
Note: you may need to restart the kernel to use updated packages.


In [7]:
from MyCapytain.resources.texts.local.capitains.cts import CapitainsCtsText

with open("../tei/tlg0525.tlg001.perseus-grc2.xml") as f:
    text = CapitainsCtsText(urn="urn:cts:greekLit:tlg0525.tlg001.perseus-grc2", resource=f)

In [8]:
from lxml import etree
from MyCapytain.common.constants import Mimetypes

urns = []
raw_xmls = []
unannotated_strings = []

for ref in text.getReffs(level=len(text.citation)):
    urn = f"{text.urn}:{ref}"
    node = text.getTextualNode(ref)
    raw_xml = node.export(Mimetypes.XML.TEI)
    tree = node.export(Mimetypes.PYTHON.ETREE)
    s = etree.tostring(tree, encoding="unicode", method="text")

    urns.append(urn)
    raw_xmls.append(raw_xml)
    unannotated_strings.append(s)

In [9]:
%pip install numpy==1.26.4

%pip install pandas


[notice] A new release of pip is available: 24.0 -> 24.2
[notice] To update, run: pip install --upgrade pip
Note: you may need to restart the kernel to use updated packages.

[notice] A new release of pip is available: 24.0 -> 24.2
[notice] To update, run: pip install --upgrade pip
Note: you may need to restart the kernel to use updated packages.


In [10]:
import pandas as pd

d = {
    "urn": pd.Series(urns, dtype="string"),
    "raw_xml": raw_xmls,
    "unannotated_strings": pd.Series(unannotated_strings, dtype="string")
}
pausanias_df = pd.DataFrame(d)

In [11]:
%pip install spacy
%pip install grecy


[notice] A new release of pip is available: 24.0 -> 24.2
[notice] To update, run: pip install --upgrade pip
Note: you may need to restart the kernel to use updated packages.

[notice] A new release of pip is available: 24.0 -> 24.2
[notice] To update, run: pip install --upgrade pip
Note: you may need to restart the kernel to use updated packages.


In [12]:
%pip show grecy

Name: grecy
Version: 1.0
Summary: grecy installs six ancient Greek spaCy models that were trained using the Universal Dependency Proiel and Perseus treebanks.
Home-page: 
Author: 
Author-email: Jacobo Myerston <jmyerston@ucsd.edu>
License: MIT License

Copyright (c) 2023 Jacobo Myerston

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
F

In [13]:
%pip show spacy

Name: spacy
Version: 3.7.6
Summary: Industrial-strength Natural Language Processing (NLP) in Python
Home-page: https://spacy.io
Author: Explosion
Author-email: contact@explosion.ai
License: MIT
Location: /workspaces/intro-text-analysis/.venv/lib/python3.12/site-packages
Requires: catalogue, cymem, jinja2, langcodes, murmurhash, numpy, packaging, preshed, pydantic, requests, setuptools, spacy-legacy, spacy-loggers, srsly, thinc, tqdm, typer, wasabi, weasel
Required-by: grc_proiel_sm, grecy
Note: you may need to restart the kernel to use updated packages.


In [14]:
%pip show pip

Name: pip
Version: 24.0
Summary: The PyPA recommended tool for installing Python packages.
Home-page: 
Author: 
Author-email: The pip developers <distutils-sig@python.org>
License: MIT
Location: /workspaces/intro-text-analysis/.venv/lib/python3.12/site-packages
Requires: 
Required-by: 
Note: you may need to restart the kernel to use updated packages.


In [15]:
%pip install pip==24.0


[notice] A new release of pip is available: 24.0 -> 24.2
[notice] To update, run: pip install --upgrade pip
Note: you may need to restart the kernel to use updated packages.


In [16]:
%run -m grecy install grc_proiel_sm


Installing grc_proiel_sm.....

Please wait, this could take some minutes.....

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0.0/65.5 MB ? eta -:--:--
     ━━━╺━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.5/65.5 MB 164.3 MB/s eta 0:00:01
     ━━━━━━╸━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 11.2/65.5 MB 171.5 MB/s eta 0:00:01
     ━━━━━━━━━╺━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 15.5/65.5 MB 144.1 MB/s eta 0:00:01
     ━━━━━━━━━━━╸━━━━━━━━━━━━━━━━━━━━━━━━━━━ 19.4/65.5 MB 122.9 MB/s eta 0:00:01
     ━━━━━━━━━━━━━━╸━━━━━━━━━━━━━━━━━━━━━━━━ 24.9/65.5 MB 127.2 MB/s eta 0:00:01
     ━━━━━━━━━━━━━━━━━━╺━━━━━━━━━━━━━━━━━━━━ 30.5/65.5 MB 162.0 MB/s eta 0:00:01
     ━━━━━━━━━━━━━━━━━━━━━╸━━━━━━━━━━━━━━━━━ 36.4/65.5 MB 167.7 MB/s eta 0:00:01
     ━━━━━━━━━━━━━━━━━━━━━━━╸━━━━━━━━━━━━━━━ 39.5/65.5 MB 131.3 MB/s eta 0:00:01
     ━━━━━━━━━━━━━━━━━━━━━━━━━╸━━━━━━━━━━━━━ 42.9/65.5 MB 106.3 MB/s eta 0:00:01
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━╺━━━━━━━━━━━ 46.4/65.5 MB 89.8 MB/s eta 0:00:01
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╺━━

In [17]:
import spacy

nlp = spacy.load("grc_proiel_sm", disable=["ner"])

In [18]:
tokenizer = nlp.tokenizer

pausanias_df['tokens'] = pausanias_df['unannotated_strings'].apply(tokenizer)

In [19]:
pausanias_df['whitespaced_tokens'] = pausanias_df['unannotated_strings'].str.split()

pausanias_df

urn  \
0     urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:1...   
1     urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:1...   
2     urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:1...   
3     urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:1...   
4     urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:1...   
...                                                 ...   
3165  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:1...   
3166  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:1...   
3167  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:1...   
3168  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:1...   
3169  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:1...   

                                                raw_xml  \
0     <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
1     <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
2     <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
3     <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
4     <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
...                                                 ...   
3165  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
3166  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
3167  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
3168  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
3169  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   

                                    unannotated_strings  \
0     τῆς ἠπείρου τῆς Ἑλληνικῆς κατὰ νήσους τὰς Κυκλ...   
1     ὁ δὲ Πειραιεὺς δῆμος μὲν ἦν ἐκ παλαιοῦ, πρότερ...   
2     θέας δὲ ἄξιον τῶν ἐν Πειραιεῖ μάλιστα Ἀθηνᾶς ἐ...   
3     ἔστι δὲ καὶ ἄλλος Ἀθηναίοις ὁ μὲν ἐπὶ Μουνυχίᾳ...   
4     ἀπέχει δὲ σταδίους εἴκοσιν ἄκρα Κωλιάς· ἐς ταύ...   
...                                                 ...   
3165  οὗτοι μὲν δὴ ὑπεροικοῦσιν Ἀμφίσσης· ἐπὶ θαλάσσ...   
3166  κληθῆναι δὲ ἀπὸ γυναικὸς ἢ νύμφης τεκμαίρομαι ...   
3167  τὰ δὲ ἔπη τὰ Ναυπάκτια ὀνομαζόμενα ὑπὸ Ἑλλήνων...   
3168  ἐνταῦθα ἔστι μὲν ἐπὶ θαλάσσῃ ναὸς Ποσειδῶνος κ...   
3169  τοῦ δὲ Ἀσκληπιοῦ τὸ
ἱερὸν ἐρείπια ἦν, ἐξ ἀρχῆς...   

                                                 tokens  \
0     (τῆς, ἠπείρου, τῆς, Ἑλληνικῆς, κατὰ, νήσους, τ...   
1     (ὁ, δὲ, Πειραιεὺς, δῆμος, μὲν, ἦν, ἐκ, παλαιοῦ...   
2     (θέας, δὲ, ἄξιον, τῶν, ἐν, Πειραιεῖ, μάλιστα, ...   
3     (ἔστι, δὲ, καὶ, ἄλλος, Ἀθηναίοις, ὁ, μὲν, ἐπὶ,...   
4     (ἀπέχει, δὲ, σταδίους, εἴκοσιν, ἄκρα, Κωλιάς, ...   
...                                                 ...   
3165  (οὗτοι, μὲν, δὴ, ὑπεροικοῦσιν, Ἀμφίσσης, ·, ἐπ...   
3166  (κληθῆναι, δὲ, ἀπὸ, γυναικὸς, ἢ, νύμφης, τεκμα...   
3167  (τὰ, δὲ, ἔπη, τὰ, Ναυπάκτια, ὀνομαζόμενα, ὑπὸ,...   
3168  (ἐνταῦθα, ἔστι, μὲν, ἐπὶ, θαλάσσῃ, ναὸς, Ποσει...   
3169  (τοῦ, δὲ, Ἀσκληπιοῦ, τὸ, \n, ἱερὸν, ἐρείπια, ἦ...   

                                     whitespaced_tokens  
0     [τῆς, ἠπείρου, τῆς, Ἑλληνικῆς, κατὰ, νήσους, τ...  
1     [ὁ, δὲ, Πειραιεὺς, δῆμος, μὲν, ἦν, ἐκ, παλαιοῦ...  
2     [θέας, δὲ, ἄξιον, τῶν, ἐν, Πειραιεῖ, μάλιστα, ...  
3     [ἔστι, δὲ, καὶ, ἄλλος, Ἀθηναίοις, ὁ, μὲν, ἐπὶ,...  
4     [ἀπέχει, δὲ, σταδίους, εἴκοσιν, ἄκρα, Κωλιάς·,...  
...                                                 ...  
3165  [οὗτοι, μὲν, δὴ, ὑπεροικοῦσιν, Ἀμφίσσης·, ἐπὶ,...  
3166  [κληθῆναι, δὲ, ἀπὸ, γυναικὸς, ἢ, νύμφης, τεκμα...  
3167  [τὰ, δὲ, ἔπη, τὰ, Ναυπάκτια, ὀνομαζόμενα, ὑπὸ,...  
3168  [ἐνταῦθα, ἔστι, μὲν, ἐπὶ, θαλάσσῃ, ναὸς, Ποσει...  
3169  [τοῦ, δὲ, Ἀσκληπιοῦ, τὸ, ἱερὸν, ἐρείπια, ἦν,, ...  

[3170 rows x 5 columns]

In [20]:
pausanias_df['whitespaced_tokens'].explode().count()

217416

In [21]:
len(pausanias_df['whitespaced_tokens'].explode().unique())

41363

In [22]:
from collections import Counter

type_counts = Counter(pausanias_df['whitespaced_tokens'].explode())

type_counts.most_common(100)

[('καὶ', 11810),
 ('δὲ', 10616),
 ('ἐς', 4162),
 ('τοῦ', 3392),
 ('τὸ', 3262),
 ('ἐν', 3114),
 ('τὴν', 2940),
 ('τε', 2791),
 ('μὲν', 2791),
 ('τῶν', 2375),
 ('τὸν', 2316),
 ('τῆς', 2299),
 ('ὁ', 1978),
 ('οἱ', 1976),
 ('τὰ', 1931),
 ('ἐπὶ', 1908),
 ('τῷ', 1877),
 ('τῇ', 1575),
 ('ὡς', 1117),
 ('τοῖς', 1061),
 ('τοὺς', 1032),
 ('ἐκ', 1018),
 ('ὑπὸ', 1000),
 ('δὴ', 975),
 ('ἐστιν', 962),
 ('ἡ', 942),
 ('οὐ', 916),
 ('γὰρ', 874),
 ('κατὰ', 802),
 ('πρὸς', 779),
 ('ἀπὸ', 776),
 ('εἶναι', 641),
 ('παρὰ', 633),
 ('οὐκ', 631),
 ('τὰς', 522),
 ('ἔτι', 520),
 ('δέ', 494),
 ('ἢ', 472),
 ('ἐξ', 455),
 ('ἐστὶν', 454),
 ('αὐτὸν', 446),
 ('αὐτῷ', 444),
 ('ἐνταῦθα', 442),
 ('ἦν', 441),
 ('ἔστι', 402),
 ('μάλιστα', 393),
 ('τοῦτο', 388),
 ('ἱερὸν', 363),
 ('ἄγαλμα', 361),
 ('ἐστι', 358),
 ('μετὰ', 358),
 ('σφισιν', 328),
 ('ὕστερον', 328),
 ('αὐτῶν', 328),
 ('περὶ', 319),
 ('οὖν', 319),
 ('ταῖς', 305),
 ('ὄνομα', 295),
 ('γε', 275),
 ('γενέσθαι', 275),
 ('τότε', 274),
 ('αἱ', 274),
 ('ὅτι', 270),
 ('

In [23]:
types = [t.text for t in pausanias_df['tokens'].explode() if not t.is_stop and t.is_alpha]

type_counts = Counter(types)

type_counts.most_common(100)


[('ἐνταῦθα', 491),
 ('μάλιστα', 421),
 ('ἄγαλμα', 411),
 ('ἱερὸν', 369),
 ('ὕστερον', 351),
 ('σφισιν', 347),
 ('ὄνομα', 345),
 ('γενέσθαι', 329),
 ('λέγουσιν', 328),
 ('τότε', 287),
 ('ἐπʼ', 268),
 ('πόλιν', 258),
 ('ἤδη', 257),
 ('σφᾶς', 250),
 ('φασιν', 238),
 ('δʼ', 234),
 ('πρότερον', 227),
 ('σφίσιν', 216),
 ('παῖδα', 215),
 ('ἐγένετο', 203),
 ('πεποίηται', 199),
 ('ὕδωρ', 196),
 ('λέγουσι', 194),
 ('Λακεδαιμονίων', 190),
 ('ἐφʼ', 186),
 ('ἔχει', 184),
 ('αὖθις', 183),
 ('ἱερόν', 183),
 ('λίθου', 183),
 ('Ἀθηνᾶς', 178),
 ('θεῶν', 176),
 ('Ἀθηναίων', 168),
 ('λέγεται', 167),
 ('Ἀπόλλωνος', 166),
 ('ἀγάλματα', 164),
 ('Ἑλλήνων', 164),
 ('σφισι', 162),
 ('δύο', 162),
 ('πρὸ', 160),
 ('φασὶν', 159),
 ('ἔνθα', 158),
 ('ὁμοῦ', 154),
 ('ἅτε', 154),
 ('ἀρχῆς', 153),
 ('Διὸς', 151),
 ('Ἀθηναίοις', 149),
 ('Ἀχαιῶν', 146),
 ('πολὺ', 142),
 ('Ἀρτέμιδος', 139),
 ('ἔργον', 138),
 ('ἀρχαῖον', 138),
 ('ἐπίκλησιν', 135),
 ('ἀπʼ', 134),
 ('γῆν', 134),
 ('Λακεδαιμόνιοι', 134),
 ('παίδων', 132),
 ('

In [24]:
raw_texts = [t for t in pausanias_df['unannotated_strings']]
annotated_texts = nlp.pipe(raw_texts, batch_size=100)

pausanias_df['nlp_docs'] = list(annotated_texts)

In [25]:
lemmata = [t.lemma_ for t in pausanias_df['nlp_docs'].explode() if not t.is_stop and t.is_alpha]

lemmata_counts = Counter(lemmata)

lemmata_counts.most_common(100)

[('ποιέω', 1201),
 ('λέγω', 1185),
 ('ἔχω', 1152),
 ('σφεῖς', 1093),
 ('γίγνομαι', 1090),
 ('παῖς', 827),
 ('καλέω', 797),
 ('πόλις', 778),
 ('φημί', 765),
 ('πολύς', 742),
 ('ἄγαλμα', 697),
 ('ἀνήρ', 678),
 ('ἱερόν', 657),
 ('θεός', 644),
 ('Λακεδαιμόνιος', 574),
 ('ἐνταῦθα', 525),
 ('λόγος', 482),
 ('πᾶς', 473),
 ('ἐπί', 454),
 ('μάλα', 449),
 ('ναός', 445),
 ('ὄνομα', 434),
 ('γυνή', 393),
 ('Ἀθηναῖος', 383),
 ('ὅσος', 380),
 ('ὕστερος', 376),
 ('γῆ', 371),
 ('Μεσσήνιος', 370),
 ('μέγας', 369),
 ('Ἀπόλλων', 359),
 ('Ἕλλην', 358),
 ('ὀνομάζω', 351),
 ('πρῶτος', 350),
 ('ἀρχή', 345),
 ('ἀφικνέομαι', 337),
 ('Ζεύς', 327),
 ('στάδιον', 327),
 ('Ἠλεῖος', 321),
 ('ποταμός', 320),
 ('ἄγω', 319),
 ('ἄνθρωπος', 316),
 ('ἔργον', 311),
 ('ἀρχαῖος', 308),
 ('πρότερος', 306),
 ('πόλεμος', 303),
 ('Ἀχαιός', 298),
 ('τότε', 291),
 ('θάλασσα', 288),
 ('Ἀργεῖος', 285),
 ('μάχη', 281),
 ('ὄρος', 273),
 ('Ἀρκάς', 272),
 ('λίθος', 272),
 ('ὕδωρ', 267),
 ('ἕτερος', 265),
 ('Ἄρτεμις', 262),
 ('χώρα', 257

In [26]:
def get_book_of_pausanias(df: pd.DataFrame, book_n: int):
    return df[df['urn'].str.startswith(f"urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:{book_n}")]

Book 1

In [27]:
get_book_of_pausanias(pausanias_df, 1)


urn  \
0     urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:1...   
1     urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:1...   
2     urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:1...   
3     urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:1...   
4     urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:1...   
...                                                 ...   
3165  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:1...   
3166  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:1...   
3167  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:1...   
3168  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:1...   
3169  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:1...   

                                                raw_xml  \
0     <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
1     <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
2     <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
3     <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
4     <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
...                                                 ...   
3165  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
3166  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
3167  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
3168  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
3169  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   

                                    unannotated_strings  \
0     τῆς ἠπείρου τῆς Ἑλληνικῆς κατὰ νήσους τὰς Κυκλ...   
1     ὁ δὲ Πειραιεὺς δῆμος μὲν ἦν ἐκ παλαιοῦ, πρότερ...   
2     θέας δὲ ἄξιον τῶν ἐν Πειραιεῖ μάλιστα Ἀθηνᾶς ἐ...   
3     ἔστι δὲ καὶ ἄλλος Ἀθηναίοις ὁ μὲν ἐπὶ Μουνυχίᾳ...   
4     ἀπέχει δὲ σταδίους εἴκοσιν ἄκρα Κωλιάς· ἐς ταύ...   
...                                                 ...   
3165  οὗτοι μὲν δὴ ὑπεροικοῦσιν Ἀμφίσσης· ἐπὶ θαλάσσ...   
3166  κληθῆναι δὲ ἀπὸ γυναικὸς ἢ νύμφης τεκμαίρομαι ...   
3167  τὰ δὲ ἔπη τὰ Ναυπάκτια ὀνομαζόμενα ὑπὸ Ἑλλήνων...   
3168  ἐνταῦθα ἔστι μὲν ἐπὶ θαλάσσῃ ναὸς Ποσειδῶνος κ...   
3169  τοῦ δὲ Ἀσκληπιοῦ τὸ
ἱερὸν ἐρείπια ἦν, ἐξ ἀρχῆς...   

                                                 tokens  \
0     (τῆς, ἠπείρου, τῆς, Ἑλληνικῆς, κατὰ, νήσους, τ...   
1     (ὁ, δὲ, Πειραιεὺς, δῆμος, μὲν, ἦν, ἐκ, παλαιοῦ...   
2     (θέας, δὲ, ἄξιον, τῶν, ἐν, Πειραιεῖ, μάλιστα, ...   
3     (ἔστι, δὲ, καὶ, ἄλλος, Ἀθηναίοις, ὁ, μὲν, ἐπὶ,...   
4     (ἀπέχει, δὲ, σταδίους, εἴκοσιν, ἄκρα, Κωλιάς, ...   
...                                                 ...   
3165  (οὗτοι, μὲν, δὴ, ὑπεροικοῦσιν, Ἀμφίσσης, ·, ἐπ...   
3166  (κληθῆναι, δὲ, ἀπὸ, γυναικὸς, ἢ, νύμφης, τεκμα...   
3167  (τὰ, δὲ, ἔπη, τὰ, Ναυπάκτια, ὀνομαζόμενα, ὑπὸ,...   
3168  (ἐνταῦθα, ἔστι, μὲν, ἐπὶ, θαλάσσῃ, ναὸς, Ποσει...   
3169  (τοῦ, δὲ, Ἀσκληπιοῦ, τὸ, \n, ἱερὸν, ἐρείπια, ἦ...   

                                     whitespaced_tokens  \
0     [τῆς, ἠπείρου, τῆς, Ἑλληνικῆς, κατὰ, νήσους, τ...   
1     [ὁ, δὲ, Πειραιεὺς, δῆμος, μὲν, ἦν, ἐκ, παλαιοῦ...   
2     [θέας, δὲ, ἄξιον, τῶν, ἐν, Πειραιεῖ, μάλιστα, ...   
3     [ἔστι, δὲ, καὶ, ἄλλος, Ἀθηναίοις, ὁ, μὲν, ἐπὶ,...   
4     [ἀπέχει, δὲ, σταδίους, εἴκοσιν, ἄκρα, Κωλιάς·,...   
...                                                 ...   
3165  [οὗτοι, μὲν, δὴ, ὑπεροικοῦσιν, Ἀμφίσσης·, ἐπὶ,...   
3166  [κληθῆναι, δὲ, ἀπὸ, γυναικὸς, ἢ, νύμφης, τεκμα...   
3167  [τὰ, δὲ, ἔπη, τὰ, Ναυπάκτια, ὀνομαζόμενα, ὑπὸ,...   
3168  [ἐνταῦθα, ἔστι, μὲν, ἐπὶ, θαλάσσῃ, ναὸς, Ποσει...   
3169  [τοῦ, δὲ, Ἀσκληπιοῦ, τὸ, ἱερὸν, ἐρείπια, ἦν,, ...   

                                               nlp_docs  
0     (τῆς, ἠπείρου, τῆς, Ἑλληνικῆς, κατὰ, νήσους, τ...  
1     (ὁ, δὲ, Πειραιεὺς, δῆμος, μὲν, ἦν, ἐκ, παλαιοῦ...  
2     (θέας, δὲ, ἄξιον, τῶν, ἐν, Πειραιεῖ, μάλιστα, ...  
3     (ἔστι, δὲ, καὶ, ἄλλος, Ἀθηναίοις, ὁ, μὲν, ἐπὶ,...  
4     (ἀπέχει, δὲ, σταδίους, εἴκοσιν, ἄκρα, Κωλιάς, ...  
...                                                 ...  
3165  (οὗτοι, μὲν, δὴ, ὑπεροικοῦσιν, Ἀμφίσσης, ·, ἐπ...  
3166  (κληθῆναι, δὲ, ἀπὸ, γυναικὸς, ἢ, νύμφ

In [39]:
get_book_of_pausanias(pausanias_df, 1)['whitespaced_tokens'].explode().count()

51265

In [54]:
len(get_book_of_pausanias(pausanias_df, 1)['whitespaced_tokens'].explode().unique())

14656

In [29]:
types = [t.text for t in get_book_of_pausanias(pausanias_df, 1)['tokens'].explode() if not t.is_stop and t.is_alpha]

type_counts = Counter(types)

type_counts.most_common(100)


[('ἐνταῦθα', 127),
 ('μάλιστα', 99),
 ('ὕστερον', 95),
 ('λέγουσιν', 89),
 ('ἄγαλμα', 86),
 ('γενέσθαι', 83),
 ('ὄνομα', 81),
 ('Ἀθηναίοις', 80),
 ('σφισιν', 80),
 ('σφᾶς', 75),
 ('πόλιν', 74),
 ('τότε', 69),
 ('Ἀθηναίων', 68),
 ('ἐπʼ', 66),
 ('ἔχει', 63),
 ('πρότερον', 58),
 ('δʼ', 58),
 ('λέγεται', 57),
 ('παῖδα', 56),
 ('ἤδη', 55),
 ('Ἀθηνᾶς', 54),
 ('λέγουσι', 53),
 ('σφίσιν', 52),
 ('ἱερὸν', 52),
 ('πλησίον', 51),
 ('Ἀπόλλωνος', 50),
 ('Ἀθηναῖοι', 49),
 ('Ἑλλήνων', 48),
 ('φασιν', 47),
 ('ἔργον', 43),
 ('ἔνθα', 41),
 ('πεποίηται', 41),
 ('σφισι', 40),
 ('Φωκέων', 40),
 ('αὖθις', 39),
 ('θεοῦ', 39),
 ('θεῶν', 38),
 ('πρῶτον', 38),
 ('ὁμοῦ', 38),
 ('ἱερόν', 36),
 ('ὅσον', 36),
 ('πρὶν', 35),
 ('γῆν', 35),
 ('ἀγάλματα', 34),
 ('ἐφʼ', 34),
 ('διʼ', 34),
 ('ὕδωρ', 34),
 ('ἀρχῆς', 33),
 ('καλούμενον', 33),
 ('πάντα', 33),
 ('ἐγένετο', 33),
 ('ἐποίησεν', 32),
 ('Ἀπόλλωνα', 32),
 ('παρʼ', 32),
 ('Μακεδόνων', 31),
 ('ὁμοίως', 31),
 ('χαλκοῦν', 31),
 ('πόλει', 31),
 ('ναυσὶν', 30),
 ('φασὶν

In [30]:
lemmata = [t.lemma_ for t in get_book_of_pausanias(pausanias_df, 1)['nlp_docs'].explode() if not t.is_stop and t.is_alpha]

lemmata_counts = Counter(lemmata)

lemmata_counts.most_common(100)

[('λέγω', 305),
 ('ἔχω', 286),
 ('σφεῖς', 280),
 ('ποιέω', 269),
 ('γίγνομαι', 219),
 ('πολύς', 196),
 ('πόλις', 194),
 ('καλέω', 190),
 ('παῖς', 186),
 ('φημί', 180),
 ('Ἀθηναῖος', 161),
 ('θεός', 154),
 ('ἀνήρ', 148),
 ('ἄγαλμα', 143),
 ('πᾶς', 136),
 ('ἐνταῦθα', 134),
 ('Ἀπόλλων', 130),
 ('Ἕλλην', 111),
 ('ἱερόν', 109),
 ('Δελφοί', 109),
 ('ὅσος', 108),
 ('λόγος', 108),
 ('πρῶτος', 107),
 ('ὄνομα', 107),
 ('μάλα', 104),
 ('ναός', 103),
 ('ὕστερος', 100),
 ('γυνή', 100),
 ('ἐπί', 100),
 ('ἔργον', 94),
 ('γῆ', 93),
 ('Θησεύς', 82),
 ('Ἀθήναιος', 80),
 ('ἀποθνῄσκω', 80),
 ('γράφω', 78),
 ('πρότερος', 77),
 ('μέγας', 76),
 ('πόλεμος', 75),
 ('ἄνθρωπος', 74),
 ('ἀρχή', 73),
 ('τότε', 73),
 ('ἀφικνέομαι', 73),
 ('μάχη', 72),
 ('ἀρχαῖος', 72),
 ('ὀνομάζω', 71),
 ('Ἀθήνη', 70),
 ('Μακεδών', 70),
 ('βάρβαρος', 70),
 ('ἄγω', 69),
 ('ναῦς', 66),
 ('θυγάτηρ', 65),
 ('ἔρχομαι', 64),
 ('θάλασσα', 63),
 ('χάλκεος', 62),
 ('βασιλεύς', 61),
 ('καθίστημι', 60),
 ('ποταμός', 60),
 ('χώρα', 58),
 ('οἶδ

Book 2

In [31]:
get_book_of_pausanias(pausanias_df, 2)

urn  \
311  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:2...   
312  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:2...   
313  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:2...   
314  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:2...   
315  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:2...   
..                                                 ...   
617  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:2...   
618  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:2...   
619  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:2...   
620  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:2...   
621  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:2...   

                                               raw_xml  \
311  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
312  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
313  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
314  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
315  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
..                                                 ...   
617  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
618  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
619  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
620  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
621  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   

                                   unannotated_strings  \
311  ἡ δὲ Κορινθία χώρα μοῖρα οὖσα τῆς Ἀργείας ἀπὸ ...   
312  Κόρινθον δὲ οἰκοῦσι Κορινθίων μὲν οὐδεὶς ἔτι τ...   
313  τῆς δὲ Κορινθίας ἐστὶ γῆς καὶ ὁ καλούμενος Κρο...   
314  ἔστι δὲ ἐπὶ τοῦ ἰσθμοῦ τῆς ἀρχῆς, ἔνθα ὁ λῃστὴ...   
315  καθήκει δὲ ὁ τῶν Κορινθίων ἰσθμὸς τῇ μὲν ἐς τὴ...   
..                                                 ...   
617  οὗτος μὲν δή σφισιν ἐκ τελετῆς, ἣν ἄγουσι τῇ Ἥ...   
618  ἔστι δὲ ἐκ Λέρνης καὶ ἑτέρα παρʼ αὐτὴν ὁδὸς τὴ...   
619  δὲ ἄνω πρὸς τὴν ἤπειρον ἀπʼ αὐτῆς χωρίον ἐστίν...   
620  ἀπὸ δὲ τῶν
πολυανδρίων ἰόντι Ἀνθήνη τέ ἐστιν, ...   
621  ἀνατείνει δὲ ὑπὲρ τὰς κώμας ὄρος Πάρνων, καὶ Λ...   

                                                tokens  \
311  (ἡ, δὲ, Κορινθία, χώρα, μοῖρα, οὖσα, τῆς, Ἀργε...   
312  (Κόρινθον, δὲ, οἰκοῦσι, Κορινθίων, μὲν, οὐδεὶς...   
313  (τῆς, δὲ, Κορινθίας, ἐστὶ, γῆς, καὶ, ὁ, καλούμ...   
314  (ἔστι, δὲ, ἐπὶ, τοῦ, ἰσθμοῦ, τῆς, ἀρχῆς, ,, ἔν...   
315  (καθήκει, δὲ, ὁ, τῶν, Κορινθίων, ἰσθμὸς, τῇ, μ...   
..                                                 ...   
617  (οὗτος, μὲν, δή, σφισιν, ἐκ, τελετῆς, ,, ἣν, ἄ...   
618  (ἔστι, δὲ, ἐκ, Λέρνης, καὶ, ἑτέρα, παρʼ, αὐτὴν...   
619  (δὲ, ἄνω, πρὸς, τὴν, ἤπειρον, ἀπʼ, αὐτῆς, χωρί...   
620  (ἀπὸ, δὲ, τῶν, \n, πολυανδρίων, ἰόντι, Ἀνθήνη,...   
621  (ἀνατείνει, δὲ, ὑπὲρ, τὰς, κώμας, ὄρος, Πάρνων...   

                                    whitespaced_tokens  \
311  [ἡ, δὲ, Κορινθία, χώρα, μοῖρα, οὖσα, τῆς, Ἀργε...   
312  [Κόρινθον, δὲ, οἰκοῦσι, Κορινθίων, μὲν, οὐδεὶς...   
313  [τῆς, δὲ, Κορινθίας, ἐστὶ, γῆς, καὶ, ὁ, καλούμ...   
314  [ἔστι, δὲ, ἐπὶ, τοῦ, ἰσθμοῦ, τῆς, ἀρχῆς,, ἔνθα...   
315  [καθήκει, δὲ, ὁ, τῶν, Κορινθίων, ἰσθμὸς, τῇ, μ...   
..                                                 ...   
617  [οὗτος, μὲν, δή, σφισιν, ἐκ, τελετῆς,, ἣν, ἄγο...   
618  [ἔστι, δὲ, ἐκ, Λέρνης, καὶ, ἑτέρα, παρʼ, αὐτὴν...   
619  [δὲ, ἄνω, πρὸς, τὴν, ἤπειρον, ἀπʼ, αὐτῆς, χωρί...   
620  [ἀπὸ, δὲ, τῶν, πολυανδρίων, ἰόντι, Ἀνθήνη, τέ,...   
621  [ἀνατείνει, δὲ, ὑπὲρ, τὰς, κώμας, ὄρος, Πάρνων...   

                                              nlp_docs  
311  (ἡ, δὲ, Κορινθία, χώρα, μοῖρα, οὖσα, τῆς, Ἀργε...  
312  (Κόρινθον, δὲ, οἰκοῦσι, Κορινθίων, μὲν, οὐδεὶς...  
313  (τῆς, δὲ, Κορινθίας, ἐστὶ, γῆς, καὶ, ὁ, καλούμ...  
314  (ἔστι, δὲ, ἐπὶ, τοῦ, ἰσθμοῦ, τῆς, ἀρχῆς, ,, ἔν...  
315  (καθήκει, δὲ, ὁ, τῶν, Κορινθίων, ἰσθμὸς, τῇ, μ...  
..                                                 ...  
617  (οὗτος, μὲν, δή, σφισιν, ἐκ, τελετῆς, ,, ἣν, ἄ...  
618  (ἔστι, δὲ, ἐκ, Λέρνης, καὶ, ἑτέρα, παρʼ, αὐτὴν...  
619  (δὲ, ἄνω, πρὸς, τὴν, ἤπειρον, ἀπʼ, αὐτῆς, χωρί..

In [40]:
get_book_of_pausanias(pausanias_df, 2)['whitespaced_tokens'].explode().count()

22835

In [55]:
len(get_book_of_pausanias(pausanias_df, 2)['whitespaced_tokens'].explode().unique())

7515

In [33]:
types = [t.text for t in get_book_of_pausanias(pausanias_df, 2)['tokens'].explode() if not t.is_stop and t.is_alpha]

type_counts = Counter(types)

type_counts.most_common(100)


[('ἐνταῦθα', 82),
 ('ἱερὸν', 74),
 ('λέγουσιν', 65),
 ('ἄγαλμα', 59),
 ('μάλιστα', 45),
 ('ἱερόν', 40),
 ('γενέσθαι', 38),
 ('ὕστερον', 36),
 ('φασιν', 36),
 ('παῖδα', 35),
 ('ἐφʼ', 34),
 ('ὄνομα', 33),
 ('λίθου', 33),
 ('ἐπʼ', 32),
 ('Ἀργείων', 32),
 ('δʼ', 31),
 ('Ἀπόλλωνος', 31),
 ('ναός', 30),
 ('ναὸς', 30),
 ('ὕδωρ', 30),
 ('σφισιν', 29),
 ('Ποσειδῶνος', 28),
 ('Ἀργεῖοι', 28),
 ('ἀγάλματα', 27),
 ('λέγουσι', 26),
 ('νῦν', 26),
 ('πεποίηται', 26),
 ('Ἀθηνᾶς', 26),
 ('μνῆμα', 25),
 ('Ἀρτέμιδος', 25),
 ('ἀπʼ', 24),
 ('πρότερον', 24),
 ('πλησίον', 24),
 ('Διὸς', 23),
 ('ἀρχῆς', 23),
 ('γῆς', 23),
 ('θάλασσαν', 23),
 ('ἐντὸς', 23),
 ('Ἥρας', 23),
 ('ἤδη', 23),
 ('πρῶτον', 22),
 ('τότε', 22),
 ('πλὴν', 21),
 ('Ἀσκληπιοῦ', 21),
 ('πρὸ', 21),
 ('σφίσιν', 20),
 ('Ἀφροδίτης', 20),
 ('ἐπίκλησιν', 20),
 ('ἔχει', 20),
 ('χωρίον', 20),
 ('σφᾶς', 19),
 ('σφισι', 19),
 ('γῆν', 19),
 ('Δήμητρος', 19),
 ('ξόανον', 18),
 ('Διονύσου', 18),
 ('ἐποίησεν', 18),
 ('φασὶν', 18),
 ('ὄρος', 18),
 ('ἔνθα', 1

In [34]:
lemmata = [t.lemma_ for t in get_book_of_pausanias(pausanias_df, 2)['nlp_docs'].explode() if not t.is_stop and t.is_alpha]

lemmata_counts = Counter(lemmata)

lemmata_counts.most_common(100)

[('λέγω', 182),
 ('ποιέω', 146),
 ('ἱερόν', 132),
 ('ἔχω', 119),
 ('καλέω', 117),
 ('γίγνομαι', 111),
 ('φημί', 109),
 ('ναός', 105),
 ('παῖς', 103),
 ('Ἀργεῖος', 99),
 ('σφεῖς', 98),
 ('ἄγαλμα', 97),
 ('ἐνταῦθα', 87),
 ('θεός', 76),
 ('πολύς', 67),
 ('ἐπί', 66),
 ('ὀνομάζω', 65),
 ('γυνή', 61),
 ('πόλις', 60),
 ('γῆ', 56),
 ('ἀρχή', 56),
 ('Ἀπόλλων', 54),
 ('ἀνήρ', 51),
 ('Ζεύς', 50),
 ('λίθος', 49),
 ('Ἄρτεμις', 47),
 ('Ποσειδῶν', 46),
 ('μάλα', 46),
 ('ἕτερος', 45),
 ('πᾶς', 44),
 ('Ἄργος', 44),
 ('θάλασσα', 43),
 ('ὄνομα', 39),
 ('ἔρχομαι', 39),
 ('μνῆμα', 39),
 ('λόγος', 39),
 ('πρῶτος', 38),
 ('ὕστερος', 38),
 ('ὁδός', 37),
 ('ἄγω', 36),
 ('ὕδωρ', 36),
 ('Ἀσκληπιός', 35),
 ('ὄρος', 35),
 ('Δημήτηρ', 35),
 ('θυγάτηρ', 34),
 ('ἀρχαῖος', 34),
 ('βασιλεύω', 34),
 ('Ἐπιδαύριος', 34),
 ('οἶδα', 33),
 ('ἀποθνῄσκω', 33),
 ('ἐκεῖνος', 33),
 ('χώρα', 32),
 ('πρότερος', 32),
 ('Ἀθήνη', 32),
 ('Ἥρα', 32),
 ('Κορίνθιος', 31),
 ('ἀπό', 31),
 ('δέ', 31),
 ('Σικυώνιος', 31),
 ('Διόνυσος', 31),
 

Book 3

In [35]:
get_book_of_pausanias(pausanias_df, 3)

urn  \
622  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:3...   
623  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:3...   
624  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:3...   
625  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:3...   
626  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:3...   
..                                                 ...   
890  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:3...   
891  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:3...   
892  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:3...   
893  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:3...   
894  urn:cts:greekLit:tlg0525.tlg001.perseus-grc2:3...   

                                               raw_xml  \
622  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
623  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
624  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
625  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
626  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
..                                                 ...   
890  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
891  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
892  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
893  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   
894  <TEI xmlns="http://www.tei-c.org/ns/1.0" xmlns...   

                                   unannotated_strings  \
622  μετὰ δὲ τοὺς Ἑρμᾶς ἐστιν ἤδη Λακωνικὴ τὰ πρὸς ...   
623  ἅτε δὲ οὐκ ὄντων· αὐτῷ παίδων ἀρρένων βασιλεύε...   
624  Ἀμύκλας δὲ ὁ Λακεδαίμονος, βουλόμενος ὑπολιπέσ...   
625  Κυνόρτα δὲ ἐγένετο Οἴβαλος. οὗτος Γοργοφόνην τ...   
626  χρόνῳ δὲ ὕστερον κατῆλθέ τε ὑπὸ Ἡρακλέους Τυνδ...   
..                                                 ...   
890  Καρδαμύλη δέ, ἧς καὶ Ὅμηρος μνήμην ἐποιήσατο ἐ...   
891  πόλιν δὲ ὀνομαζομένην ἐν τοῖς ἔπεσιν Ἐνόπην το...   
892  ἐνταῦθα ἐν τῇ Γερηνίᾳ Μαχάονος τοῦ Ἀσκληπιοῦ μ...   
893  διὸ καὶ τάδε αὐτὸς οἶδα περὶ τὸ Ἀσκληπιεῖον τὸ...   
894  τῆς δὲ χώρας τῆς Γερηνίας ὄρος Καλάθιόν ἐστιν·...   

                                                tokens  \
622  (μετὰ, δὲ, τοὺς, Ἑρμᾶς, ἐστιν, ἤδη, Λακωνικὴ, ...   
623  (ἅτε, δὲ, οὐκ, ὄντων, ·, αὐτῷ, παίδων, ἀρρένων...   
624  (Ἀμύκλας, δὲ, ὁ, Λακεδαίμονος, ,, βουλόμενος, ...   
625  (Κυνόρτα, δὲ, ἐγένετο, Οἴβαλος, ., οὗτος, Γοργ...   
626  (χρόνῳ, δὲ, ὕστερον, κατῆλθέ, τε, ὑπὸ, Ἡρακλέο...   
..                                                 ...   
890  (Καρδαμύλη, δέ, ,, ἧς, καὶ, Ὅμηρος, μνήμην, ἐπ...   
891  (πόλιν, δὲ, ὀνομαζομένην, ἐν, τοῖς, ἔπεσιν, Ἐν...   
892  (ἐνταῦθα, ἐν, τῇ, Γερηνίᾳ, Μαχάονος, τοῦ, Ἀσκλ...   
893  (διὸ, καὶ, τάδε, αὐτὸς, οἶδα, περὶ, τὸ, Ἀσκληπ...   
894  (τῆς, δὲ, χώρας, τῆς, Γερηνίας, ὄρος, Καλάθιόν...   

                                    whitespaced_tokens  \
622  [μετὰ, δὲ, τοὺς, Ἑρμᾶς, ἐστιν, ἤδη, Λακωνικὴ, ...   
623  [ἅτε, δὲ, οὐκ, ὄντων·, αὐτῷ, παίδων, ἀρρένων, ...   
624  [Ἀμύκλας, δὲ, ὁ, Λακεδαίμονος,, βουλόμενος, ὑπ...   
625  [Κυνόρτα, δὲ, ἐγένετο, Οἴβαλος., οὗτος, Γοργοφ...   
626  [χρόνῳ, δὲ, ὕστερον, κατῆλθέ, τε, ὑπὸ, Ἡρακλέο...   
..                                                 ...   
890  [Καρδαμύλη, δέ,, ἧς, καὶ, Ὅμηρος, μνήμην, ἐποι...   
891  [πόλιν, δὲ, ὀνομαζομένην, ἐν, τοῖς, ἔπεσιν, Ἐν...   
892  [ἐνταῦθα, ἐν, τῇ, Γερηνίᾳ, Μαχάονος, τοῦ, Ἀσκλ...   
893  [διὸ, καὶ, τάδε, αὐτὸς, οἶδα, περὶ, τὸ, Ἀσκληπ...   
894  [τῆς, δὲ, χώρας, τῆς, Γερηνίας, ὄρος, Καλάθιόν...   

                                              nlp_docs  
622  (μετὰ, δὲ, τοὺς, Ἑρμᾶς, ἐστιν, ἤδη, Λακωνικὴ, ...  
623  (ἅτε, δὲ, οὐκ, ὄντων, ·, αὐτῷ, παίδων, ἀρρένων...  
624  (Ἀμύκλας, δὲ, ὁ, Λακεδαίμονος, ,, βουλόμενος, ...  
625  (Κυνόρτα, δὲ, ἐγένετο, Οἴβαλος, ., οὗτος, Γοργ...  
626  (χρόνῳ, δὲ, ὕστερον, κατῆλθέ, τε, ὑπὸ, Ἡρακλέο...  
..                                                 ...  
890  (Καρδαμύλη, δέ, ,, ἧς, καὶ, Ὅμηρος, μνήμην, ἐπ...  
891  (πόλιν, δὲ, ὀνομαζομένην, ἐν, τοῖς, ἔπεσιν, Ἐν...  
892  (ἐνταῦθα, ἐν, τῇ, Γερηνίᾳ, Μαχάονος, τοῦ, Ἀσκλ..

In [41]:
get_book_of_pausanias(pausanias_df, 3)['whitespaced_tokens'].explode().count()

17737

In [56]:
len(get_book_of_pausanias(pausanias_df, 3)['whitespaced_tokens'].explode().unique())

6132

In [37]:
types = [t.text for t in get_book_of_pausanias(pausanias_df, 3)['tokens'].explode() if not t.is_stop and t.is_alpha]

type_counts = Counter(types)

type_counts.most_common(100)


[('ἄγαλμα', 58),
 ('ἱερὸν', 52),
 ('ἐνταῦθα', 45),
 ('Λακεδαιμόνιοι', 38),
 ('λέγουσιν', 36),
 ('μάλιστα', 36),
 ('Λακεδαιμονίων', 36),
 ('ὕστερον', 35),
 ('Λακεδαιμονίοις', 35),
 ('σφισιν', 34),
 ('ἱερόν', 32),
 ('φασιν', 31),
 ('Ἀθηνᾶς', 29),
 ('ὄνομα', 27),
 ('πεποίηται', 26),
 ('ἤδη', 25),
 ('τότε', 24),
 ('ἐγένετο', 22),
 ('γενέσθαι', 22),
 ('Ἡρακλέους', 22),
 ('ἐπʼ', 22),
 ('χωρίον', 22),
 ('σταδίους', 22),
 ('πρότερον', 21),
 ('Ἀπόλλωνος', 21),
 ('παῖδα', 20),
 ('παῖδας', 20),
 ('Ἀσκληπιοῦ', 20),
 ('Σπάρτῃ', 19),
 ('Διὸς', 19),
 ('Ἀρτέμιδος', 19),
 ('θεῶν', 19),
 ('φασὶν', 18),
 ('ἐπίκλησιν', 18),
 ('σφίσιν', 17),
 ('ἐφʼ', 17),
 ('Ἀθηναίων', 17),
 ('ἀρχαῖον', 17),
 ('ὕδωρ', 16),
 ('ἀρχὴν', 16),
 ('δʼ', 16),
 ('ἔνθα', 16),
 ('προελθόντι', 16),
 ('λέγουσι', 15),
 ('θάλασσαν', 15),
 ('πόλιν', 15),
 ('θαλάσσῃ', 15),
 ('Ἀθηναίοις', 15),
 ('αὐτόθι', 15),
 ('ναός', 15),
 ('ναὸς', 15),
 ('ξόανον', 15),
 ('οἶδα', 15),
 ('ἀφίκετο', 14),
 ('Τυνδάρεω', 14),
 ('γῆν', 14),
 ('πολὺ', 14),
 ('μ

In [38]:
lemmata = [t.lemma_ for t in get_book_of_pausanias(pausanias_df, 3)['nlp_docs'].explode() if not t.is_stop and t.is_alpha]

lemmata_counts = Counter(lemmata)

lemmata_counts.most_common(100)

[('Λακεδαιμόνιος', 125),
 ('ποιέω', 102),
 ('λέγω', 100),
 ('γίγνομαι', 94),
 ('ἱερόν', 93),
 ('σφεῖς', 85),
 ('φημί', 83),
 ('ἔχω', 82),
 ('καλέω', 79),
 ('ἄγαλμα', 76),
 ('παῖς', 72),
 ('πολύς', 58),
 ('θεός', 58),
 ('πόλις', 57),
 ('ἐνταῦθα', 50),
 ('ναός', 45),
 ('θάλασσα', 44),
 ('στάδιον', 44),
 ('ἀνήρ', 43),
 ('ὀνομάζω', 42),
 ('πόλεμος', 41),
 ('ἀρχή', 40),
 ('Σπάρτη', 40),
 ('ἐπί', 39),
 ('Ἀπόλλων', 37),
 ('ὕστερος', 37),
 ('μάλα', 37),
 ('λόγος', 36),
 ('γυνή', 35),
 ('μάχη', 34),
 ('Ἀθήνη', 34),
 ('ὄνομα', 32),
 ('βασιλεύς', 32),
 ('Ἄρτεμις', 32),
 ('ἄγω', 32),
 ('Ἀθηναῖος', 31),
 ('ἀρχαῖος', 31),
 ('ἀφικνέομαι', 30),
 ('πρότερος', 29),
 ('Μεσσήνιος', 29),
 ('Ἡρακλῆς', 29),
 ('Ἀργεῖος', 29),
 ('Ἀγησίλαος', 29),
 ('παρέχω', 28),
 ('Κλεομένης', 28),
 ('βασιλεύω', 27),
 ('χωρίον', 27),
 ('Ζεύς', 26),
 ('ὁδός', 26),
 ('ἤδη', 25),
 ('πρῶτος', 25),
 ('γῆ', 25),
 ('ἕτερος', 25),
 ('θυγάτηρ', 24),
 ('τότε', 24),
 ('μνῆμα', 24),
 ('προέρχομαι', 24),
 ('Ἆγις', 24),
 ('οἶδα', 24),
 ('Π

Analysis/Commentary: 

Right away, I noticed that the first three Books of Pausanias differ quite a bit in terms of length -- i.e., token count. Book 1 has the greatest token count at 51,265 tokens, which is significantly more than Book 2's 22,835 tokens, which is in turn greater than Book 3's 17,737 tokens. I find it interesting that Book 1 is so much larger than than Books 2 and 3; I am not yet very familiar with  Pausanias' work, so perhaps this difference in token count would make sense in light of the content of the Books. 

I think the data regarding the most common lemmata in each Books is interesting. Many of the same words appear in all three, especially 'ἐνταῦθα.' Given that the work is a description of various places throughout ancient Greece, it's not surprising that ἐνταῦθα would be mentioned so many times. I think this data has the potential to be extremely helpful to students, as it gives us a sense of which vocabularly is most important to learn in order to best understand and read the text. However, some lemmata are prevalent only in certain books, which may give us insight into the book's contents. For example, the lemma 'Ἀθηναῖος' and its forms appear 161 times in Book 1; 'Ἀργεῖος' and its forms appear 99 times in Book 2; and 'Λακεδαιμόνιος' and its forms appear 125 times in Book 3. Based on this data, it seems that each Book may focus primarily on a different area of Greece, with Book 1 describing Athens, Book 2 Argos, and Book 3 Sparta. I am curious to see if this turns out to be true once I've read the text. 

I have chosen to take a closer look at Book 3, in English translation for ease of reading. I expected to see many variations of terms related to the Lacedaemonians (Spartans), and that largely panned out. Throughout the entire third Book, terms such as Lacedaemon (both referring to a person and later to the city-state) and Lacedaemonians were exceedingly common; in ancient Greek these would all be variations of the lemma 'Λακεδαιμόνιος'. I was correct in my hunch that Book 3 would focus on Sparta, and if other books in the Periegesis follow a similar format, I bet they each focus on a different part of Greece as I suspected.  

Although Book 3 was about Lacedaemon/Sparta as I hypothesized, the actual content of the chapter was not quite what I had expected. Based on what little I knew about the Periegesis, I was expecting more of Book 3 to describe the city-state and its surroundings more. While it certainly does do this, it isn't really until 3.11, when Pausanias describes the market-place in Sparta; he then goes on to describe other parts of the city-state as well as its monuments. The first ten chapters of Book 3 instead ended up being largely dedicated to describing the mytho-history of the Lacedaemonians. I expected some of this, but thought that it would have been smaller than it ended up being. Even when he was describing the city-state and its monuments, he included much commentary on their cultural/mytho-history -- again more than I was expecting. Before taking a closer look at Book 3, I thought more emphasis would be placed on eyewitness accounts of monuments, locations, geography, etc. -- the physical aspects of ancient Greece -- but I realized that he focused much more on the culture and history. This was a pleasant surprise, as I found the commentary on the mytho-history throughout more interesting than the physical descriptions. 

My main take-away from my analysis is that although a quick lemmata count and analysis can yield helpful insights, the actual content may differ -- or be presented differently -- from what is expected. But perhaps I should have expected more emphasis on the cultural history of the people in the first place given that the lemma was 'Λακεδαιμόνιος' instead of 'Λακεδαίμων.' This was my first time attempting an analysis of this kind. I enjoyed it and am interested in looking both more at the Periegesis (and its potential as a pedagogical text for Greek language study) as well as other classical texts as we proceed through the course. 
